apps/submission/io/archive.py

```diff
@@ -95,8 +98,33 @@ def save(self, pixeler):
             pixeler=pixeler,
             completed_at=self.meta['analysis']['date'],
         )
-        analysis.secondary_data.name = self.meta['analysis']['secondary_data_path'].name  # noqa
-        analysis.notebook.name = self.meta['analysis']['notebook_path'].name
+
+        # Copy archive files to the media tree
+        relative_dest = Path(
+            Analysis.secondary_data_upload_to(
+                analysis,
+                self.meta['analysis']['secondary_data_path'].name
+            )
+        )
+        dest = Path(settings.MEDIA_ROOT) / relative_dest
```

Comment:
these three linesc(from here to 111) are repeated multiple times. maybe a helper receiving `relative_dest` and returning (absolute) `dest` would make sense.

Reply:
I am not disappointed: I was hesitating to do so and decided to let your review track down this! 😛👏